### DISCUSSION TITLE
Custom object return type & integration with `graphql-scalars`

### DISCUSSION THREAD

**Discussion by valerii15298** [MichalLytek/typegraphql-prisma#390] [Q&A] [answered]
I have generated resolver which looks like this: 
```ts
@TypeGraphQL.InputType("EntityUpdateInput", {
  isAbstract: true
})
export class EntityUpdateInput {
  @TypeGraphQL.Field(_type => String, {
    nullable: true
  })
  color?: string | undefined | null;
}
```

But I wanna change return type to be `HexColorCodeDefinition` from `graphql-scalars`. There are [examples](https://the-guild.dev/graphql/scalars/docs/usage/apollo-server) how to use `graphql-scalars` with apollo server but I cannot figure it out how to use it with type-graphql especially with `typegraphql-prisma`. Are there ways to integrate `graphql-scalars` with `typegraphql`?
```ts
import { HexColorCodeDefinition, HexColorCodeResolver} from 'graphql-scalars';
@TypeGraphQL.InputType("EntityUpdateInput", {
  isAbstract: true
})
export class EntityUpdateInput {
  @TypeGraphQL.Field(_type => HexColorType/*? where to get it ?*/, {
    nullable: true
  })
  color?: string | undefined | null;
}
```

I see there is [possibility](https://typegraphql.com/docs/scalars.html#custom-scalars) to create custom scalars, but it also says: 
> First of all, we need to create our own GraphQLScalarType instance *or import a scalar type from a 3rd-party npm library*. For example, Mongo's ObjectId:

So maybe there is some recommended way or example how to use `graphql-scalars` with `type-graphql`?

Also I checked [graphql-custom-types](https://github.com/stylesuxx/graphql-custom-types) as described [here](https://typegraphql.com/docs/validation.html#scalars) but `graphql-custom-types` is not as full featured and as good maintained as `graphql-scalars` package.

**Comment by MichalLytek** [comment]
May be related to #43

**Reply by valerii15298** [reply]
Thanks, @MichalLytek! I should have search for existing issues:)

**Comment by valerii15298** [comment, marked as answer]
For the input type I did find a solution to omit specific fields from input using `TypeGraphQL.omit(input: true)` in `schema.prisma`:
```prisma
model someModel {
    prop1 String
    /// TypeGraphQL.omit(input: true)
    prop2 String
    @@id(fields: [prop1, prop2])
}
```
and then extend from generated input type and add custom fields with scalars from `graphql-scalars` package:
```ts
export class GeneratedInput {
  @TypeGraphQL.Field(_type => TypeGraphQL.Int, {
    nullable: false
  })
  prop1!: number;
}
```
custom:
```ts
import { URLResolver } from 'graphql-scalars';
export class CustomInput extends GeneratedInput {
  @TypeGraphQL.Field(_type => URLResolver, {
    nullable: false
  })
  prop2: URL;
}
```

But in cases for validation, we do not necessary need that because we can use `class-validator` as @MichalLytek suggested https://github.com/MichalLytek/typegraphql-prisma/issues/43
